sdk/src/fee_calculator.rs

```diff
@@ -9,6 +9,45 @@ pub struct FeeCalculator {
     // cluster processing load.
     pub lamports_per_signature: u64,
 
+    // What portion of collected fees are to be destroyed, as a fraction of std::u8::MAX
+    pub burn_percent: u8,
+}
+
+impl Default for FeeCalculator {
+    fn default() -> Self {
+        Self {
+            lamports_per_signature: 0,
+            burn_percent: DEFAULT_BURN_PERCENT,
+        }
+    }
+}
+
+impl FeeCalculator {
+    pub fn new(lamports_per_signature: u64) -> Self {
+        Self {
+            lamports_per_signature,
+            ..FeeCalculator::default()
+        }
+    }
+
+    pub fn calculate_fee(&self, message: &Message) -> u64 {
+        self.lamports_per_signature * u64::from(message.header.num_required_signatures)
+    }
+
+    /// calculate unburned fee from a fee total, returns (unburned, burned)
+    pub fn burn(&self, fees: u64) -> (u64, u64) {
+        let burned = fees * u64::from(self.burn_percent) / 100;
+        (fees - burned, burned)
+    }
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct FeeRateGovernor {
+    // The current cost of a signature  This amount may increase/decrease over time based on
+    // cluster processing load.
+    pub lamports_per_signature: u64,
```

Comment:
`lamports_per_signature` in `FeeRateGovernor` doesn't feel right anymore.    The current fee would be held in a 
`FeeCalculator` now no?

Reply:
Yes but, the last fee rate is used to calculate the next.  I suppose I could pass the current `FeeCalculator` in with it to `new_derived()`, but would have to return the `FeeCalculator` there too.  Might be making more trouble than it's worth.  Could drop `pub`?

Reply:
dropping `pub` wfm